src/ocean/Assets.ts

```diff
@@ -239,82 +239,96 @@ export class Assets extends Instantiable {
     return (await this.ocean.metadatacache.queryMetadata(searchQuery)).results
   }
 
+  /**    Metadata updates
+   *  Don't forget to call ocean.OnChainMetadataCache.update after using this functions
+   * ie:  ocean.OnChainMetadataCache.update(ddo.id,ddo,account.getId())
+   */
+
   /**
-   * Edit Metadata for a DDO.
-   * @param  {did} string DID.
+   * Edit Metadata for a DID.
+   * @param  {did} string DID. You can leave this empty if you already have the DDO
+   * @param  {ddo} DDO if empty, will trigger a retrieve
    * @param  {newMetadata}  EditableMetadata Metadata fields & new values.
    * @param  {Account} account Ethereum account of owner to sign and prove the ownership.
-   * @return {Promise<string>}
+   * @return {Promise<DDO>} the new DDO
    */
   public async editMetadata(
     did: string,
-    newMetadata: EditableMetadata,
-    account: Account
+    ddo: DDO,
+    newMetadata: EditableMetadata
   ): Promise<DDO> {
-    const oldDdo = await this.ocean.metadatacache.retrieveDDO(did)
-    let i
-    for (i = 0; i < oldDdo.service.length; i++) {
-      if (oldDdo.service[i].type === 'metadata') {
-        if (newMetadata.title) oldDdo.service[i].attributes.main.name = newMetadata.title
-        if (!oldDdo.service[i].attributes.additionalInformation)
-          oldDdo.service[i].attributes.additionalInformation = Object()
+    if (!ddo) ddo = await this.ocean.metadatacache.retrieveDDO(did)
```

Comment:
we should decide if this method receives DID or DDO, not allow both. I would stick to only DID since it prevents accidental wrong DDO retrieval's by the user which might get passed. This would then also reduce the logic within the method a bit.

Reply:
if we have the DID, then we fetch the DDO from aqua.  but it we have the ddo, then use that in order to avoid an unnecessary call to aqua (most likely market already has the ddo)

Reply:
solved, using ddo only